### PR TITLE
Implementation of VNS-LPRC

### PR DESCRIPTION
**Pull request by Jonathan-Fontaine44**

### CODE REVIEW

File: src/vns_lprc.jl [hunk 1]
```diff
@@ -0,0 +1,227 @@
+#=
+# VNS_LPRC for the RENAULT Roadef 2005 challenge
+# inspired by work of
+# Celso C. Ribeiro, Daniel Aloise, Thiago F. Noronha, Caroline Rocha, Sebastián Urrutia
+#
+# @Author Boualem Lamraoui, Benoît Le Badezet, Benoit Loger, Jonathan Fontaine, Killian Fretaud, Rémi Garcia
+# =#
```
Comment (remi-garcia) [inline src/vns_lprc.jl]: Use the new header

File: src/vns_lprc.jl [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+include("parser.jl")
+include("solution.jl")
+include("functions.jl")
+include("constants.jl")
```
Comment (remi-garcia) [inline src/vns_lprc.jl]: Drop those include and add `include("vns_lprc.jl")` in `main.jl`

File: src/vns_lprc.jl [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            end
+            if list != []
+                k = rand(list)
+                move_exchange!(solution, i, k, instance)
+            end
+        end
+        if phi == cost_VNS_LPRC(solution, instance)
+            improved = false
+        end
+    end
```
Comment (remi-garcia) [inline src/vns_lprc.jl:120]: add `return solution`

File: src/vns_lprc.jl [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            if list != []
+                k = rand(list)
+                move!(solution, i, k, instance)
+            end
+        end
+        nb_non_improved += 1
+        if phi < cost_VNS_LPRC(solution, instance)
+            nb_non_improved = 0
+        end
+    end
```
Comment (remi-garcia) [inline src/vns_lprc.jl:153]: add `return solution`

File: src/vns_lprc.jl [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+end
+
+# Apply two local search, first one with insertion move, and the second one with exchange move.
+function intensification_VNS_LPRC!(solution::Solution, instance::Instance)
+    localSearch_intensification_VNS_LPRC!(solution, VNS_LPRC_ALPHA_PERTURBATION, cost_move_insertion, move_insertion!, instance)
+    localSearch_intensification_VNS_LPRC!(solution, VNS_LPRC_ALPHA_PERTURBATION, cost_move_exchange, move_exchange!, instance)
```
Comment (remi-garcia) [inline src/vns_lprc.jl:160]: add `return solution`

**Review by remi-garcia — CHANGES_REQUESTED** [review]
I will take more time to read this tomorrow.

File: src/vns_lprc.jl (hunk 5 above)
Comment (Jonathan-Fontaine44) [reply src/vns_lprc.jl:160]: Why should I ? Intensification is supposed to increase the current solution, not to return an other one I think.

File: src/vns_lprc.jl (hunk 2 above)
Comment (Jonathan-Fontaine44) [reply src/vns_lprc.jl]: Then this file couldn't run by it self, main shouldn't call files dependency for other files.

Comment (Jonathan-Fontaine44) [reply src/vns_lprc.jl]: I checked in the [documentation](https://docs.julialang.org/en/v1/manual/code-loading/) and a file is evaluated each times an `include` call it. So i will delete those include.

File: src/vns_lprc.jl (hunk 5 above)
Comment (remi-garcia) [reply src/vns_lprc.jl:160]: This is just something we can add. See [this](https://docs.julialang.org/en/v1/manual/style-guide/index.html#Append-!-to-names-of-functions-that-modify-their-arguments-1): _It is typical for such functions to also return the modified array for convenience._

**Review by k-fretaud — DISMISSED** [review]

**Review by remi-garcia — APPROVED** [review]
When this is merge we need an issue to remember to comment `src/vns_lprc.jl`

**Review by k-fretaud — APPROVED** [review]

**Pull request merged**